task/oci-upload/0.1/README.md

```diff
@@ -0,0 +1,145 @@
+# OCI Object Storage Upload
+
+This Tekton task uploads files or directories to Oracle Cloud Infrastructure (OCI) Object Storage buckets using the OCI CLI.
+
+## Description
+
+The `oci-upload` task allows to upload files or directories from a workspace to an OCI Object Storage bucket. It supports both single file uploads and bulk directory uploads with optional recursive directory traversal.
+
+## Prerequisites
+
+Before using this task, ensure to have:
+
+1. **Kubernetes cluster with Tekton Pipelines installed**
+2. **OCI Account with:**
+   - Tenancy OCID and User OCID
+   - API Key and Fingerprint
+   - Region identifier (e.g., `us-ashburn-1`)
+   - Object Storage namespace
+3. **Existing OCI Object Storage bucket**
+
+## Parameters
+
+| Parameter | Description | Type | Default |
+|-----------|-------------|------|---------|
+| `path` | The path to files or directories relative to the source workspace | string | **required** |
+| `bucketName` | The name of the OCI Object Storage bucket (without oci:// prefix) | string | **required** |
+| `objectPrefix` | The object prefix/path within the bucket where files should be uploaded | string | `""` |
+| `tenancyOcid` | The OCID of the OCI tenancy | string | **required** |
+| `userOcid` | The OCID of the OCI user | string | **required** |
+| `region` | The OCI region identifier | string | **required** |
+| `namespace` | The OCI Object Storage namespace (typically your tenancy name) | string | **required** |
+| `replaceExistingFiles` | When "false", files that already exist are skipped | string | `"true"` |
+| `recursive` | When "true", upload directories recursively | string | `"true"` |
+| `privateKeyPath` | The path inside the credentials workspace to the OCI API private key file | string | `oci_api_key.pem` |
+| `fingerprintPath` | The path inside the credentials workspace to the file containing the fingerprint | string | `fingerprint` |
```

Comment:
NT: We could have an optional parameter for extra args to be passed to the upload command, for instance for adding metadata or controlling parallel upload settings.